src/color.rs

```diff
@@ -0,0 +1,522 @@
+    swizzle!(b, a, r, b);
+    swizzle!(b, a, r, a);
+    swizzle!(b, a, g, r);
+    swizzle!(b, a, g, g);
+    swizzle!(b, a, g, b);
+    swizzle!(b, a, g, a);
+    swizzle!(b, a, b, r);
+    swizzle!(b, a, b, g);
+    swizzle!(b, a, b, b);
+    swizzle!(b, a, b, a);
+    swizzle!(b, a, a, r);
+    swizzle!(b, a, a, g);
+    swizzle!(b, a, a, b);
+    swizzle!(b, a, a, a);
+    swizzle!(a, r, r, r);
+    swizzle!(a, r, r, g);
+    swizzle!(a, r, r, b);
+    swizzle!(a, r, r, a);
+    swizzle!(a, r, g, r);
+    swizzle!(a, r, g, g);
+    swizzle!(a, r, g, b);
+    swizzle!(a, r, g, a);
+    swizzle!(a, r, b, r);
+    swizzle!(a, r, b, g);
+    swizzle!(a, r, b, b);
+    swizzle!(a, r, b, a);
+    swizzle!(a, r, a, r);
+    swizzle!(a, r, a, g);
+    swizzle!(a, r, a, b);
+    swizzle!(a, r, a, a);
+    swizzle!(a, g, r, r);
+    swizzle!(a, g, r, g);
+    swizzle!(a, g, r, b);
+    swizzle!(a, g, r, a);
+    swizzle!(a, g, g, r);
+    swizzle!(a, g, g, g);
+    swizzle!(a, g, g, b);
+    swizzle!(a, g, g, a);
+    swizzle!(a, g, b, r);
+    swizzle!(a, g, b, g);
+    swizzle!(a, g, b, b);
+    swizzle!(a, g, b, a);
+    swizzle!(a, g, a, r);
+    swizzle!(a, g, a, g);
+    swizzle!(a, g, a, b);
+    swizzle!(a, g, a, a);
+    swizzle!(a, b, r, r);
+    swizzle!(a, b, r, g);
+    swizzle!(a, b, r, b);
+    swizzle!(a, b, r, a);
+    swizzle!(a, b, g, r);
+    swizzle!(a, b, g, g);
+    swizzle!(a, b, g, b);
+    swizzle!(a, b, g, a);
+    swizzle!(a, b, b, r);
+    swizzle!(a, b, b, g);
+    swizzle!(a, b, b, b);
+    swizzle!(a, b, b, a);
+    swizzle!(a, b, a, r);
+    swizzle!(a, b, a, g);
+    swizzle!(a, b, a, b);
+    swizzle!(a, b, a, a);
+    swizzle!(a, a, r, r);
+    swizzle!(a, a, r, g);
+    swizzle!(a, a, r, b);
+    swizzle!(a, a, r, a);
+    swizzle!(a, a, g, r);
+    swizzle!(a, a, g, g);
+    swizzle!(a, a, g, b);
+    swizzle!(a, a, g, a);
+    swizzle!(a, a, b, r);
+    swizzle!(a, a, b, g);
+    swizzle!(a, a, b, b);
+    swizzle!(a, a, b, a);
+    swizzle!(a, a, a, r);
+    swizzle!(a, a, a, g);
+    swizzle!(a, a, a, b);
+    swizzle!(a, a, a, a);
+}
+
+impl_op_ex!(+= |a: &mut Color, b: &Color| { a.r += b.r; a.g += b.g; a.b += b.b; a.a += b.a; });
+impl_op_ex!(-= |a: &mut Color, b: &Color| { a.r -= b.r; a.g -= b.g; a.b -= b.b; a.a -= b.a; });
+
+impl_op_ex!(+ |a: &Color, b: &Color| -> Color { Color{r: a.r + b.r, g: a.g + b.g, b: a.b + b.b, a: a.a + b.a } });
+impl_op_ex!(-|a: &Color, b: &Color| -> Color {
+    Color {
+        r: a.r - b.r,
+        g: a.g - b.g,
+        b: a.b - b.b,
+        a: a.a - b.a,
+    }
+});
+
+impl From<[f32; 3]> for Color {
+    fn from(d: [f32; 3]) -> Self {
+        Self {
+            r: d[0],
+            g: d[1],
+            b: d[2],
+            a: 1.0,
+        }
+    }
+}
+
+impl From<[f32; 4]> for Color {
+    fn from(d: [f32; 4]) -> Self {
+        Self {
+            r: d[0],
+            g: d[1],
+            b: d[2],
+            a: d[3],
+        }
+    }
+}
+
```

Comment:
Maybe we should add the corresponding Into impls?

Reply:
sure